File: Examples/Python/Containers/MongoDB-REST/src/health/health_result_entry.py
class HealthResultEntry:
    def __init__(self, key: str, healthy: bool, duration: int, error: str = None):
        self.key = key
        self.duration = f'{duration} ms'
        self.healthy = healthy
        self.status = 'healthy' if healthy else 'unhealthy'
        self.error = error

    def to_dict(self):
        return {
            "key": self.key,
            "duration": self.duration,
            "healthy": self.healthy,
            "status": self.status,
            "error": self.error
        }

File: Examples/Rust/Containers/MongoDB-REST/src/src/health/health_result_entry.rs
use serde::{Serialize, Deserialize};

#[derive(Serialize, Deserialize)]
pub struct HealthResultEntry {
    pub key: String,
    pub healthy: bool,
    pub duration: String,
    pub status: String,
    pub error: Option<String>,
}

impl HealthResultEntry {
    pub fn new(key: String, healthy: bool, duration: u64, error: Option<String>) -> Self {
        let status = if healthy { "healthy".to_string() } else { "unhealthy".to_string() };
        let duration = format!("{} ms", duration);
        HealthResultEntry {
            key,
            healthy,
            duration,
            status,
            error,
        }
    }
}
